tests/crds/crds_suite_test.go

```diff
@@ -1,48 +1,84 @@
-	"github.com/onsi/gomega/gexec"
+	. "github.com/onsi/gomega/gexec"
 )
 
 func TestCrds(t *testing.T) {
 	RegisterFailHandler(Fail)
+	SetDefaultEventuallyTimeout(3 * time.Second)
 	RunSpecs(t, "CRDs Suite")
 }
 
 var rootNamespace string
 
 var _ = BeforeSuite(func() {
-	var ok bool
-	rootNamespace, ok = os.LookupEnv("ROOT_NAMESPACE")
-	if !ok {
-		rootNamespace = "cf"
+	rootNamespace = GetDefaultedEnvVar("ROOT_NAMESPACE", "cf")
+	Eventually(
+		kubectl("get", "namespace/"+rootNamespace),
+	).Should(Exit(0), "Could not find root namespace called %q", rootNamespace)
+})
+
+func kubectl(args ...string) *Session {
+	cmdStarter := commandstarter.NewCommandStarter()
+	return kubectlWithCustomReporter(cmdStarter, commandreporter.NewCommandReporter(), args...)
+}
+
+func kubectlApply(stdinText string, sprintfArgs ...any) *Session {
+	cmdStarter := commandstarter.NewCommandStarterWithStdin(
+		strings.NewReader(
+			fmt.Sprintf(stdinText, sprintfArgs...),
+		),
+	)
+	return kubectlWithCustomReporter(cmdStarter, commandreporter.NewCommandReporter(), "apply", "-f=-")
+}
+
+func kubectlWithCustomReporter(cmdStarter *commandstarter.CommandStarter, reporter *commandreporter.CommandReporter, args ...string) *Session {
+	request, err := cmdStarter.Start(reporter, "kubectl", args...)
+	if err != nil {
+		panic(err)
 	}
 
-	expectCommandToSucceed(kubectl("get", "namespace/"+rootNamespace),
-		"Could not find root namespace called %q", rootNamespace)
-})
+	return request
+}
 
-func kubectl(words ...string) *exec.Cmd {
-	return exec.Command("kubectl", words...)
+func GetRequiredEnvVar(envVarName string) string {
+	value, ok := os.LookupEnv(envVarName)
+	Expect(ok).To(BeTrue(), envVarName+" environment variable is required, but was not provided.")
+	return value
 }
 
-func writeToStdIn(cmd *exec.Cmd, stdinText string, sprintfArgs ...any) {
-	stdin, err := cmd.StdinPipe()
-	Expect(err).NotTo(HaveOccurred())
-	_, err = stdin.Write([]byte(fmt.Sprintf(stdinText, sprintfArgs...)))
-	Expect(err).NotTo(HaveOccurred())
-	stdin.Close()
+func GetDefaultedEnvVar(envVarName, defaultValue string) string {
+	value, ok := os.LookupEnv(envVarName)
+	if !ok {
+		return defaultValue
+	}
+	return value
 }
 
-func expectCommandToSucceed(cmd *exec.Cmd, optionalDescription ...any) {
-	session, err := gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
-	Expect(err).NotTo(HaveOccurred())
-	Eventually(session, "10s").Should(gexec.Exit(0), optionalDescription...)
+func loginAs(apiEndpoint string, skipSSL bool, user string) {
+	apiArguments := []string{"api", apiEndpoint}
+	if skipSSL {
+		apiArguments = append(apiArguments, "--skip-ssl-validation")
+	}
+
+	Eventually(cf.Cf(apiArguments...)).Should(Exit(0))
+
+	// Stdin contains username followed by 2 return carriages. First one
+	// enters the username and second one skips the org selection prompt that
+	// is presented if there is more than one org
+	loginSession := cf.CfWithStdin(bytes.NewBufferString(user+"\n\n"), "login")
```

Comment:
Out of curiosity, why do this instead of `cf auth user`?

Reply:
I wasn't sure if `cf auth` had made it into a released version of the CF cli yet, so I just copied what the smoke tests do. If it's safe to use, then I'll switch it.

Reply:
It looks like it's safe, so I'll make the change.